data/ui_scripts/language_setting/__init__.lua

```diff
@@ -0,0 +1,153 @@
+        LANG_SPANISH = "Espagnol"
+    },
+    italian = {
+        LANGUAGE_BUTTON = "Lingua",
+        LANGUAGE_BUTTON_DESC = "Cambia la tua lingua di gioco.",
+        POPUP_RESTART_REQUIRED_TITLE = "RIPARTENZA RICHIESTA",
+        POPUP_RESTART_REQUIRED_TEXT = "ATTENZIONE: per funzionare correttamente, devi avere la tua lingua nella directory zone della tua root di gioco. Hai cambiato la lingua del gioco, desideri riavviare?",
+        POPUP_NO_ZONE_FOUND_TITLE = "LINGUA NON TROVATA",
+        POPUP_NO_ZONE_FOUND_TEXT = "Non siamo in grado di trovare la cartella della lingua che hai selezionato, per evitare bug, non cambieremo nulla.",
+        LANG_ENGLISH = "Inglese",
+        LANG_FRENCH = "Francese",
+        LANG_ITALIAN = "Italiano",
+        LANG_GERMAN = "Tedesco",
+        LANG_SPANISH = "Spagnolo"
+    },
+    german = {
+        LANGUAGE_BUTTON = "Sprache",
+        LANGUAGE_BUTTON_DESC = "Ändern Sie Ihre Spielsprache.",
+        POPUP_RESTART_REQUIRED_TITLE = "NEUSTART ERFORDERLICH",
+        POPUP_RESTART_REQUIRED_TEXT = "WARNUNG: Um richtig zu funktionieren, müssen Sie Ihre Sprache im Zonenverzeichnis Ihres Spielstammverzeichnisses haben. Du hast deine Spielsprache geändert, möchtest du neu starten?",
+        POPUP_NO_ZONE_FOUND_TITLE = "SPRACHE NICHT GEFUNDEN",
+        POPUP_NO_ZONE_FOUND_TEXT = "Wir können den von Ihnen ausgewählten Sprachordner nicht finden, um Fehler zu vermeiden, werden wir nichts ändern.",
+        LANG_ENGLISH = "Englisch",
+        LANG_FRENCH = "Französisch",
+        LANG_ITALIAN = "Italienisch",
+        LANG_GERMAN = "Deutsch",
+        LANG_SPANISH = "Spanisch"
+    },
+    spanish = {
+        LANGUAGE_BUTTON = "Idioma",
+        LANGUAGE_BUTTON_DESC = "Cambia el idioma de tu juego.",
+        POPUP_RESTART_REQUIRED_TITLE = "REINICIO REQUERIDO",
+        POPUP_RESTART_REQUIRED_TEXT = "ADVERTENCIA: para que funcione correctamente, debe tener su idioma en el directorio de zone de la raíz del juego. Has cambiado el idioma de tu juego, ¿deseas reiniciar?",
+        POPUP_NO_ZONE_FOUND_TITLE = "IDIOMA NO ENCONTRADO",
+        POPUP_NO_ZONE_FOUND_TEXT = "No podemos encontrar la carpeta de idioma que ha seleccionado, para evitar errores, no cambiaremos nada.",
+        LANG_ENGLISH = "Inglés",
+        LANG_FRENCH = "Francés",
+        LANG_ITALIAN = "Italiano",
+        LANG_GERMAN = "Alemán",
+        LANG_SPANISH = "Español"
+    }
+}
+local language = "english"
+
+function get_actual_language()
+    local file_path = "h1-mod/language.txt"
+    local file = io.open(file_path, "r")
+
+    if file == nil then
+        return
+    end
+
+    local data = file:read("*a")
+    language = data
+end
+
+function set_language(value)
+    local file_path = "h1-mod/language.txt"
+    local file = io.open(file_path, "w")
+    file:write(value)
+    file:close()
+end
+
+function does_zone_folder_exists(language)
+    local isok, errstr, errcode = os.rename("zone/" .. language .. "/", "zone/" .. language .. "/")
```

Comment:
forgot to add this to the review - does renaming show if the folder actually exists or something?

if it does, consider using io::directoryexists

Reply:
we add it on the client [here](https://github.com/h1-mod/h1-mod/blob/develop/src/client/component/ui_scripting.cpp#L176) :)